backend/src/stwo/prover.rs

```diff
@@ -46,16 +52,111 @@ where
+                                // witness and constant traces need to be bit reversed
+                                let bit_reversed_fixed_columns: BTreeMap<
+                                    DegreeType,
+                                    Vec<(String, Vec<F>)>,
+                                > = fixed_columns
+                                    .iter()
+                                    .map(|(key, vec)| {
+                                        let transformed_vec: Vec<(String, Vec<F>)> = vec
+                                            .iter()
+                                            .map(|(name, slice)| {
+                                                let mut values: Vec<F> = slice.to_vec(); // Clone the slice into a Vec
+                                                bit_reverse_coset_to_circle_domain_order(
+                                                    &mut values,
+                                                ); // Apply bit reversal
+                                                (name.clone(), values) // Return the updated tuple
+                                            })
+                                            .collect(); // Collect the updated vector
+                                        (*key, transformed_vec) // Rebuild the BTreeMap with transformed vectors
+                                    })
+                                    .collect();
+
+                                let constant_trace: ColumnVec<
+                                    CircleEvaluation<B, BaseField, BitReversedOrder>,
+                                > = bit_reversed_fixed_columns
+                                    .values()
+                                    .flat_map(|vec| {
+                                        vec.iter().map(|(_name, values)| {
+                                            let values = values
+                                                .iter()
+                                                .map(|v| v.try_into_i32().unwrap().into())
+                                                .collect();
+                                            CircleEvaluation::new(domain, values)
+                                        })
+                                    })
+                                    .collect();
+
+                                // commit to the fixed columns
+                                let mut tree_builder = commitment_scheme.tree_builder();
+                                tree_builder.extend_evals(constant_trace.clone());
+                                tree_builder.commit(prover_channel);
+                                let trees = commitment_scheme.trees;
+
+                                (
+                                    size as usize,
+                                    TableProvingKey {
+                                        trees,
+                                        prover_channel: prover_channel.clone(),
```

Comment:
Why do we need to save the prover channel here? If this is the Fiat Shamir transcript, isn't it determined by starting with an empty transcript and observing the commits? Or is this done here to save prover time?